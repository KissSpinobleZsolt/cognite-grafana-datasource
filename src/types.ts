// eslint-disable-next-line max-classes-per-file
import {
  DataQueryRequest,
  DataQuery as DataQueryUI,
  TimeSeries,
  DataQuery,
  DataSourceJsonData,
} from '@grafana/data';
import { TimeSeriesResponseItem, Datapoints, Items, IdEither, Limit } from './cdf/types';

export interface CogniteQuery extends DataQuery {
  target: number | ''; // Timeseries internal Id:
  aggregation: string;
  granularity: string;
  error: string;
  label: string;
  tab: Tab;
  assetQuery: AssetQuery;
  expr: string;
  warning: string;
}

export enum Tab {
  Timeseries = 'Timeseries',
  Asset = 'Asset',
  Custom = 'Custom',
  Template = 'Template',
}

export const defaultAssetQuery: AssetQuery = {
  includeSubtrees: false,
  target: '',
};

// TODO: Investigate if "type" property is required, it is currently not defined in MyQuery
// These defaults are extracted from the old queryCtrl.ts
export const defaultQuery: Partial<CogniteQuery> = {
  target: '',
  // type: 'timeserie',
  aggregation: 'average',
  granularity: '',
  label: '',
  tab: Tab.Timeseries,
  expr: '',
  assetQuery: {
    target: '',
    includeSubtrees: false,
  },
};

/**
 * These are options configured for each DataSource instance
 */

export interface CogniteDataSourceOptions extends DataSourceJsonData {
  cogniteApiUrl?: string;
  authType: string;
  defaultRegion: string;
  cogniteProject: string;
}

export interface MySecureJsonData {
  apiKey?: string;
}

/**
 * Value that is used in the backend, but never sent over HTTP to the frontend
 */
export function isError(maybeError: DataQueryError | any): maybeError is DataQueryError {
  return (<DataQueryError>maybeError).error !== undefined;
}

/**
 * Comes from grafana, could be imported in future releases hopefully
 * @param name â€“ event name
 */
export const eventFactory = <T = undefined>(name: string): AppEvent<T> => {
  return { name };
};

export type QueryResponse = DataResponse<TimeSeries[]>;

export interface MetricDescription {
  readonly text: string;
  readonly value: number | string;
}

export type MetricFindQueryResponse = MetricDescription[];

export enum ParseType {
  Timeseries = 'Timeseries',
  Asset = 'Asset',
  Event = 'Event',
}

export interface AssetQuery {
  target: string;
  includeSubtrees: boolean;
  old?: AssetQuery;
  timeseries?: TimeSeriesResponseItem[];
  func?: string;
  templatedTarget?: string;
}

export interface InputQueryTarget extends DataQueryUI {
  target: number | '';
  aggregation: string;
  granularity: string;
  error: string;
  label: string;
  tab: Tab;
  assetQuery: AssetQuery;
  expr: string;
  warning: string;
  templateQuery: TemplateQuery;
}

export interface QueryTarget extends InputQueryTarget {
  target: number;
}

export type QueryFormat = 'json';

export type QueryOptions = DataQueryRequest<InputQueryTarget>;

export type Tuple<T> = [T, T];

/**
 * Comes from grafana, could be imported in future releases hopefully
 */
export interface AppEvent<T> {
  readonly name: string;
  payload?: T;
}

export enum HttpMethod {
  POST = 'POST',
  GET = 'GET',
  PATCH = 'PATCH',
  DELETE = 'DELETE',
}

export interface DataSourceRequestOptions {
  url: string;
  method: HttpMethod;
  retry?: number;
  requestId?: string;
  headers?: { [s: string]: string };
  silent?: boolean;
  data?: any;
}

export type SuccessResponse = {
  metadata: ResponseMetadata;
  result: DataQueryRequestResponse;
};

export type FailResponse = {
  metadata: ResponseMetadata;
  error: any;
};

export abstract class Result<T, E> {
  abstract isOk: boolean;
  abstract isErr: boolean;
}
export class Ok<T, E> implements Result<T, E> {
  value: T;

  constructor(v: T) {
    this.value = v;
  }

  isOk: boolean = true;
  isErr: boolean = false;
}

export class Err<T, E> implements Result<T, E> {
  error: E;

  constructor(error: E) {
    this.error = error;
  }

  isOk: boolean = false;
  isErr: boolean = true;
}

export type Responses<T, U> = {
  succeded: T[];
  failed: U[];
};

export interface DataQueryRequestResponse extends DataResponse<Datapoints> {
  config: {
    data: {
      aggregates: string;
      limit: number;
    };
  };
}

export interface RequestParams<DataType = any> {
  path: string;
  data: DataType;
  method: HttpMethod;
  params?: { [s: string]: any };
  requestId?: string;
  cacheTime?: string;
}

export type DataQueryError = {
  error: {
    cancelled?: boolean;
    data: {
      error?: {
        message: string;
        notFound?: string[];
      };
    };
    status: number;
  };
};

export type QueriesData = {
  items: DataQueryRequestItem[];
  target: QueryTarget;
}[];

export type ResponseMetadata = { labels: string[]; target: QueryTarget };

export interface DataQueryAlias {
  alias: string;
  id: number;
  aggregate?: string;
  granularity?: string;
}

export type DataQueryRequestItem = {
  expression?: string;
  start?: string | number;
  end?: string | number;
  limit?: number;
  granularity?: string;
  aggregates?: string[];
  id?: number;
};

export type Aggregates = Pick<CDFDataQueryRequest, 'aggregates'>;
export type Granularity = Pick<CDFDataQueryRequest, 'granularity'>;

export interface CDFDataQueryRequest {
  items: DataQueryRequestItem[];
  start?: string | number;
  end?: string | number;
  limit?: number;
  aggregates?: string[];
  granularity?: string;
}

export interface CogniteAnnotationQuery extends DataQuery {
  query?: string;
  error?: string;
}

export interface AnnotationSearchQuery {
  description: string;
  type: string;
  subtype: string;
  minStartTime: number;
  maxStartTime: number;
  minEndTime: number;
  maxEndTime: number;
  minCreatedTime: number;
  maxCreatedTime: number;
  minLastUpdatedTime: number;
  maxLastUpdatedTime: number;
  // format is {"k1": "v1", "k2": "v2"}
  metadata: string;
  assetIds: number[];
  assetSubtrees: number[];
  sort: 'startTime' | 'endTime' | 'createdTime' | 'lastUpdatedTime';
  dir: 'asc' | 'desc';
  limit: number;
  offset: 0;
}

export interface DataResponse<T> {
  data: T;
}

export type CursorResponse<T> = DataResponse<Items<T> & { nextCursor?: string }>;

export type Response<T = any> = DataResponse<{
  items: T[];
}>;

export type TimeseriesFilterQuery = {
  filter?: {
    description?: string;
    assetSubtreeIds?: IdEither[];
    assetIds?: string[];
    [s: string]: any; // so we auto-support next features
  };
  cursor?: string;
} & Limit;

export interface VariableQueryData {
  query: string;
  error?: string;
}

export interface VariableQueryProps {
  query: string;
  onChange: (query: VariableQueryData, description: string) => void;
  datasource: any;
  templateSrv: any;
}

export interface QueryRequestError {
  refId: string;
  error: string;
}

export interface QueryDatapointsWarning {
  refId: string;
  warning: string;
}

export interface TemplateQuery extends DataQuery {
  domain: string;
  domainVersion: number;
  queryText: string;
  dataPath: string;
  dataPointsPath: string;
  groupBy: string;
  aliasBy: string;
  annotationTitle: string;
  annotationText: string;
  annotationTags: string;
  constant: number;
}

export const defaultTemplateQuery: Partial<TemplateQuery> = {
  domain: undefined,
  domainVersion: undefined,
  queryText: `query {
      wellList {
        name,
        pressure {
          datapoints(start: $__from, end: $__to, limit: 50) {
            timestamp,
            value
          }
        }
      }
  }`,
  dataPath: 'data',
  dataPointsPath: '',
  groupBy: '',
  aliasBy: '',
  annotationTitle: '',
  annotationText: '',
  annotationTags: '',
  constant: 6.5,
};
