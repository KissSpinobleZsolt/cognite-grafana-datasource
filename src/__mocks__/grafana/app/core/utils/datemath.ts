// for tests, just pass in the millisecond timestamps as strings
export function parse(text: any, roundUp?: any): any {
  return Number(text);
}
