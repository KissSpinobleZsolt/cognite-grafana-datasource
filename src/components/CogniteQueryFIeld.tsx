// import _ from 'lodash';
// import React from 'react';
// import { QueryEditorProps } from '@grafana/ui';
// import CogniteDatasource from '../datasource';
// import { QueryTarget, CogniteOptions } from '../types';

// interface CogniteQueryEditorProps extends QueryEditorProps<CogniteDatasource, QueryTarget, CogniteOptions> {}

// interface CogniteQueryEditorState {}

// class CogniteQueryEditor extends React.PureComponent<CogniteQueryEditorProps, CogniteQueryEditorState> {
//   constructor(props: CogniteQueryEditorProps, context: React.Context<any>) {
//     super(props, context);
//     this.state = {};
//   }

//   componentDidMount() {}

//   componentWillUnmount() {}

//   componentDidUpdate(prevProps: CogniteQueryEditorProps) {}
// }

// export default CogniteQueryEditor;
