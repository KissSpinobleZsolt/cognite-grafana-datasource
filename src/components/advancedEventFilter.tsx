import React, { useState, useRef, useEffect, useCallback } from 'react';
import CodeMirror from 'codemirror';
import 'codemirror/lib/codemirror.css';
import '../css/dracula.css';
import 'codemirror/addon/hint/show-hint';
import 'codemirror/addon/lint/lint';
import { EventQuery } from '../types';

export const AdvancedEventFilter = (props) => {
  const { query, onQueryChange } = props;
  const textAreaRef = useRef(null);
  const [editor, setEditor] = useState<CodeMirror.EditorFromTextArea | null>(null);

  const [eventQuery, setEventQuery] = useState(query.eventQuery);
  const patchEventQuery = useCallback(
    (eventQueryPatch: Partial<EventQuery>) => {
      setEventQuery({ ...eventQuery, ...eventQueryPatch });
    },
    [eventQuery]
  );

  const triggerQuery = useCallback(() => {
    onQueryChange({
      eventQuery,
    });
  }, [eventQuery, onQueryChange]);
  useEffect(() => {
    if (textAreaRef.current) {
      const editor = CodeMirror.fromTextArea(textAreaRef.current, {
        mode: { name: 'javascript', json: true },
        theme: 'dracula',
        /* lint: {
          schema: myGraphQLSchema,
          validationRules: [ExampleRule],
        },
        hintOptions: {
          schema: myGraphQLSchema,
        }, */
      });
      setEditor(editor);
      editor.getDoc().setValue(query.eventQuery.eventQuery);
    }
  }, [textAreaRef]);

  useEffect(() => {
    if (editor != null) {
      const handleChange = () => {
        patchEventQuery({ eventQuery: editor.getDoc().getValue() });
      };

      const handleBlur = () => {
        triggerQuery();
      };

      editor.getDoc().on('change', handleChange);
      editor.on('blur', handleBlur);
      return () => {
        editor.getDoc().off('change', handleChange);
        editor.off('blur', handleBlur);
      };
    }

    return undefined;
  }, [editor, patchEventQuery, triggerQuery]);

  return (
    <div className="gf-form gf-form--grow">
      <span className="gf-form-label query-keyword fix-query-keyword width-10">Query</span>
      <textarea ref={textAreaRef} name="eventQuery" />
    </div>
  );
};
