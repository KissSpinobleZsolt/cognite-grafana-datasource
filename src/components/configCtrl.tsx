import React, { ChangeEvent, PureComponent } from 'react';
import { LegacyForms, Tooltip } from '@grafana/ui';
import { DataSourcePluginOptionsEditorProps } from '@grafana/data';
import { MyDataSourceOptions, MySecureJsonData } from '../types';

const { SecretFormField, FormField } = LegacyForms;

type Props = DataSourcePluginOptionsEditorProps<MyDataSourceOptions>;

interface State {}  // eslint-disable-line

export class ConfigEditor extends PureComponent<Props, State> {
  onPathChange = (event: ChangeEvent<HTMLInputElement>) => {
    const { onOptionsChange, options } = this.props;
    const jsonData = {
      ...options.jsonData,
      path: event.target.value,
    };
    onOptionsChange({ ...options, jsonData });
  };

  // TODO: Verify that this is correct.
  onProjectChange = (event: ChangeEvent<HTMLInputElement>) => {
    const { onOptionsChange, options } = this.props;
    const jsonData = {
      ...options.jsonData,
      project: event.target.value,
    };
    onOptionsChange({ ...options, jsonData });
  };

  // Secure field (only sent to the backend)
  onAPIKeyChange = (event: ChangeEvent<HTMLInputElement>) => {
    const { onOptionsChange, options } = this.props;
    onOptionsChange({
      ...options,
      secureJsonData: {
        apiKey: event.target.value,
      },
    });
  };

  onResetAPIKey = () => {
    const { onOptionsChange, options } = this.props;
    onOptionsChange({
      ...options,
      secureJsonFields: {
        ...options.secureJsonFields,
        apiKey: false,
      },
      secureJsonData: {
        ...options.secureJsonData,
        apiKey: '',
      },
    });
  };

  render() {
    const { options } = this.props;
    const { jsonData, secureJsonFields } = options;
    const secureJsonData = (options.secureJsonData || {}) as MySecureJsonData;
    const tooltip = `This is the URL used to reach the API.
      If the project is deployed on the default multi-tenant installation (most are),
      then keep the default value and do not change the URL.
      If the project is deployed on a separate custom cluster,
      then change the URL to point at the API server for that cluster.
      If unsure, leave the URL as default.`;

    return (
      <div className="gf-form-group">
        <div className="gf-form">
          <FormField
            label="Project"
            labelWidth={6}
            inputWidth={20}
            onChange={this.onProjectChange}
            value={jsonData.project || ''}
            placeholder="Cognite Data Fusion project"
          />
        </div>

        <div className="gf-form-inline">
          <div className="gf-form">
            <SecretFormField
              isConfigured={(secureJsonFields && secureJsonFields.apiKey) as boolean}
              value={secureJsonData.apiKey || ''}
              label="API Key"
              placeholder="Cognite Data Fusion API key"
              labelWidth={6}
              inputWidth={20}
              onReset={this.onResetAPIKey}
              onChange={this.onAPIKeyChange}
            />
          </div>
        </div>

        <div className="gf-form">
          <FormField
            label="Path"
            labelWidth={6}
            inputWidth={20}
            onChange={this.onPathChange}
            value={jsonData.path || ''}
            placeholder="api.cognitedata.com"
            tooltip={tooltip}
          />
        </div>
      </div>
    );
  }
}
