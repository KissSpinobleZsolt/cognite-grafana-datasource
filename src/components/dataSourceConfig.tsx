import React, { useState, useCallback } from 'react';
import { css, cx } from 'emotion';
import { DataSourceSettings, SelectableValue } from '@grafana/data';

import { LegacyForms } from '@grafana/ui';
import { DataSourceProps, MyDataSourceOptions } from 'types';

const { FormField, Input, Select } = LegacyForms;

// import { InlineFormLabel } from '../FormLabel/FormLabel';
// import { TagsInput } from '../TagsInput/TagsInput';
// import { useTheme } from '../../themes';

export interface HttpSettingsBaseProps {
  /** The configuration object of the data source */
  dataSourceConfig: DataSourceSettings<MyDataSourceOptions, any>;
  /** Callback for handling changes to the configuration object */
  onChange: (config: DataSourceSettings) => void;
}

export interface HttpSettingsProps extends HttpSettingsBaseProps {
  /** The default url for the data source */
  defaultUrl: string;
  /** Show the http access help box */
  showAccessOptions?: boolean;
}

const DEFAULT_ACCESS_OPTION = {
  label: 'Server (default)',
  value: 'proxy',
};

export const DataSourceConfig: React.FC<DataSourceProps> = props => {
  const { defaultUrl, defaultProject, dataSourceConfig, onChange, showAccessOptions } = props;
  const [isAccessHelpVisible, setIsAccessHelpVisible] = useState(false);
  // const theme = useTheme();

  const onSettingsChange = useCallback(
    (change: Partial<DataSourceSettings<any, any>>) => {
      onChange({
        ...dataSourceConfig,
        ...change,
      });
    },
    [dataSourceConfig]
  );

  const isValidUrl = /^(ftp|http|https):\/\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-\/]))?$/.test(
    dataSourceConfig.url
  );

  const notValidStyle = css`
    box-shadow: inset 0 0px 5px red;
  `;

  const inputStyle = cx({ [`width-20`]: true, [notValidStyle]: !isValidUrl });

  function ApiUrlInput(props) {
    const urlInput = (
      <Input
        className={props.inputStyle}
        placeholder={props.defaultUrl}
        value={props.dataSourceConfig.url}
        onChange={event => onSettingsChange({ url: event.currentTarget.value })}
      />
    );
    let tooltip =
      `This is the URL used to reach the API.
      If the project is deployed on the default multi-tenant installation (most are),
      then keep the default value and do not change the URL.
      If the project is deployed on a separate custom cluster,
      then change the URL to point at the API server for that cluster.
      If unsure, leave the URL as default.`;

    return (
      <div className="gf-form-inline">
        <div className="gf-form">
          {
            // FIXME: ctrl.current.jsonData.cogniteApiUrl
          }
          <FormField
            label="API URL"
            labelWidth={11}
            tooltip={tooltip}
            inputEl={urlInput}
          />
        </div>
      </div>
    )
  }

  function ProjectInput(props) {
    let projectInput = (
      <Input
        className={inputStyle}
        placeholder={defaultProject}
        value={dataSourceConfig.jsonData.project}
        onChange={event => onSettingsChange({ url: event.currentTarget.value })}
      />)

    return (
      <div className="gf-form-inline">
        <div className="gf-form">
          <FormField
            label="Project"
            labelWidth={11}
            tooltip="Cognite Data Fusion project name."
            inputEl={projectInput}
          />
        </div>
      </div>
    )
  }

  function ApiKeyInput(props) {
    let input;
    if (props.dataSourceConfig.url) {
      input = (
        <Input
          className={inputStyle}
          placeholder={defaultUrl}
          value={dataSourceConfig.jsonData.project}
          onChange={event => onSettingsChange({ url: event.currentTarget.value /** Set Api-key instead of URL */})}
          type="password"
        />
        /* <info-popover mode="right-absolute">
                <p>Cognite Data Fusion API key.</p>
            </info-popover> */
      );
    } else {
      input = (
        <>
          <Input
          className={inputStyle}
          placeholder="Configured"
          value={dataSourceConfig.jsonData.project}
          disabled={true}
          />
          <a
            className="btn btn-secondary gf-form-btn"
            href="#"
            onClick={/*Unset API key here?*/()=>(null)} // ctrl.current.secureJsonFields.cogniteDataPlatformApiKey
          >
            Reset
          </a>
        </>
      )
    }

    return (
      <div className="gf-form-inline">
        <div className="gf-form">
          <FormField
            label="API Key"
            labelWidth={11}
            tooltip="Cognite Data Fusion API key."
            inputEl={input}
          />
        </div>
      </div>
    )
  };

  return (
    <div>
      <h3 className="page-heading">Authentication</h3>

      <div className="gf-form-group">
        <ApiKeyInput dataSourceConfig={DataSourceConfig} />
        <ProjectInput />
        <ApiUrlInput inputStyle={inputStyle} defaultUrl={defaultUrl} dataSourceConfig={dataSourceConfig}/>
      </div>
    </div>
  );
};
