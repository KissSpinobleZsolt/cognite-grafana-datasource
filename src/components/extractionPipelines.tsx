import React, { useEffect, useState } from 'react';
import { AsyncMultiSelect, Field, Input, Switch, Tooltip } from '@grafana/ui';
import _ from 'lodash';
import CogniteDatasource from '../datasource';
import { SelectedProps } from './queryEditor';
import '../css/extraction.css';

export const ExtractionPipelinesTab = (
  props: SelectedProps & { datasource: CogniteDatasource }
) => {
  const {
    query: { extractionPipelinesQuery },
    onQueryChange,
    datasource,
  } = props;
  const { numeric, ids, enableAllExtractionPipelines } = extractionPipelinesQuery;
  return (
    <div className="extraction-pipelines">
      <Field label="Extaction Pipeline: ">
        <AsyncMultiSelect
          loadOptions={() =>
            datasource.extractionPipelinesDatasource.getAllExtractionPipelines().then((res) =>
              _.map(res, ({ name, externalId }) => ({
                label: name,
                value: externalId,
                id: externalId,
              }))
            )
          }
          value={ids}
          defaultOptions
          allowCustomValue
          placeholder="Select extraction pipeline"
          className="cognite-dropdown width-20"
          onChange={(values) =>
            onQueryChange({
              extractionPipelinesQuery: {
                ...extractionPipelinesQuery,
                ids: values,
              },
            })
          }
        />
      </Field>
      <Field label="List all Extraction Pipelines">
        <Switch
          value={enableAllExtractionPipelines}
          onChange={() =>
            onQueryChange({
              extractionPipelinesQuery: {
                ...extractionPipelinesQuery,
                enableAllExtractionPipelines: !enableAllExtractionPipelines,
              },
            })
          }
        />
      </Field>
      <Field label="Change to numeric status">
        <Switch
          value={numeric}
          onChange={() =>
            onQueryChange({
              extractionPipelinesQuery: {
                ...extractionPipelinesQuery,
                numeric: !numeric,
              },
            })
          }
        />
      </Field>
    </div>
  );
};
