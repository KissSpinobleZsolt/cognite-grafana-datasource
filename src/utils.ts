import _ from 'lodash';

export default class Utils {
  // Converts an object to a query string, ignores properties with undefined/null values
  // TODO: maybe clean this up a bit, might break easily
  static getQueryString(obj: any) {
    return _.reduce(
      obj,
      (result: string, val: any, key: string) => {
        return _.isNil(val)
          ? result
          : _.isArray(val)
          ? `${result + [key, val].map(encodeURIComponent).join('=[')}]&`
          : `${result + [key, val].map(encodeURIComponent).join('=')}&` + '';
      },
      ''
    ).slice(0, -1);
  }

  static getDatasourceValueString(aggregation: string): string {
    const mapping = {
      undefined: 'value',
      none: 'value',
      avg: 'average',
      int: 'interpolation',
      stepinterpolation: 'stepInterpolation',
      step: 'stepInterpolation',
      continuousvariance: 'continousVariance', // spelling mistake is intended - will have to change in 0.6
      continuousVariance: 'continousVariance',
      cv: 'continousVariance',
      discretevariance: 'discreteVariance',
      dv: 'discreteVariance',
      totalvariation: 'totalVariation',
      tv: 'totalVariation',
    };
    return mapping[aggregation] || aggregation;
  }
}
