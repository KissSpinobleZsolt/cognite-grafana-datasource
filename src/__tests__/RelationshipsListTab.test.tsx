import React from 'react';
import 'react-dom';
import Enzyme, { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { RelationshipsListTab } from '../components/relationshipsListTab';
import { defaultQuery, CogniteQuery } from '../types';
import { getMockedDataSource } from './utils';

Enzyme.configure({ adapter: new Adapter() });

const defaultCogniteQuery = defaultQuery as CogniteQuery;
const ds = getMockedDataSource();
const elem = (
  <RelationshipsListTab
    query={defaultCogniteQuery}
    datasource={ds}
    onQueryChange={() => jest.fn()}
  />
);
const wrapper = shallow(elem);
describe('RelationshipsListTab', () => {
  it('equals with snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  // TODO integrate a full react component unit and feature test
  /* it('renders under mount', () => {
    wrapper = mount(elem);
  }); */
});
