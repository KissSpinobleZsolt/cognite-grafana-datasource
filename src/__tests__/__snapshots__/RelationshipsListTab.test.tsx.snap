// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelationshipsListTab equals with snapshot 1`] = `
<div
  className="full-width-row"
>
  <MultiSelect
    allowCustomValue={true}
    className="cognite-dropdown"
    maxMenuHeight={150}
    onChange={[Function]}
    options={Array []}
    placeholder="Filter relations by dataset"
    value={Array []}
  />
  <MultiSelect
    allowCustomValue={true}
    className="cognite-dropdown"
    maxMenuHeight={150}
    onChange={[Function]}
    options={Array []}
    placeholder="Filter relations by Label"
    value={Array []}
  />
</div>
`;
